CONTRIBUTING.md

```diff
@@ -21,6 +21,53 @@ For your contribution to be accepted you will need to sign the [Shopify Contribu
 4. Push to the branch (`git push origin my-new-feature`)
 5. Create a new Pull Request
 
+## Run Language Server 
+
+If you're making changes to the language server and you want to debug, you can run the repo's version of theme-check-language-server
+
+### VS Code
+
+* Install the [Shopify Liquid](https://github.com/shopify/theme-check-vscode) plugin
+* At minimum add the following to `settings.json` (be sure to update `shopifyLiquid.languageServerPath` with your local path to theme-check)
+
+ ```json
+"shopifyLiquid.formatterDevPreview": true,
+"shopifyLiquid.languageServerPath": "/path/to/theme-check/bin/theme-check-language-server",
+ ```
+
+ * Setup your local environment by running the following script in your console from the root of theme-check
+
+ ```bash
+#!/usr/bin/env bash
+cd "$HOME/src/github.com/Shopify/theme-check" &> /dev/null
+
+chruby () {
+  source '/opt/dev/sh/chruby/chruby.sh'
+  chruby "$@"
+}
+
+export THEME_CHECK_DEBUG=true
+export THEME_CHECK_DEBUG_LOG_FILE="/tmp/theme-check-debug.log"
+touch "$THEME_CHECK_DEBUG_LOG_FILE"
+chruby 2.7 &>/dev/null
+gem env &>/dev/null
+
+ ```
+* In another terminal form the root of theme check run `tail -f /tmp/theme-check-debug.log` to watch the server logs 
+* Restart VS Code, confirm the response for initialize in the logs is pointing to the language server in the repo, it should look something like this:
+
+```json
+    "serverInfo": {
+        "name": "/Users/johndoe/src/github.com/Shopify/theme-check/bin/theme-check-language-server",
+        "version": "1.10.3"
+    }
+```
+
+ ### Vim
+
+ #### TODO
```

Comment:
@charlespwd how does running language server differ on vim (so i can write that up too)?

Reply:
It depends which completion engine you use, if you use `coc.nvim`, then all you need is this in your CocConfig.
```
  "languageserver": {
    "theme-check": {
      "command": "/Users/<YOUR_USERNAME>/bin/theme-check-language-server",
      "trace.server": "verbose",
      "filetypes": ["liquid", "liquid.html"],
      "rootPatterns": [".theme-check.yml", "snippets/*"],
      "settings": {
        "themeCheck": {
          "checkOnSave": true,
          "checkOnEnter": true,
          "checkOnChange": false
        }
      }
    }
```